saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/classifier/SL_model/Saul_SL_Inference.scala

```diff
@@ -95,11 +96,13 @@ class Saul_SL_Inference[HEAD <: AnyRef](factors: List[ConstrainedClassifier[_, H
             //    println("size mismatch!", cf.onClassifier.classifier.asInstanceOf[SparseNetworkLBP].getNetwork.get(i).asInstanceOf[LinearThresholdUnit].getWeightVector.size(), ",", ltuSize.size)
             //              }
             val exampleFeatureIndexes = ofDim[Int](myLTUJointlyTrainedWeight.length)
-            cf.onClassifier.classifier.asInstanceOf[SparseNetworkLBP].getLTU(i).getWeightVector.clear()
+            val weightVector = cf.onClassifier.classifier.asInstanceOf[SparseNetworkLearner].getLTU(i).asInstanceOf[LinearThresholdUnit].getWeightVector
+            weightVector.clear()
```

Comment:
Here, I had issues with making this really effect the classifier. That is why I did it in this weird way. Now I am not sure if this `clear` on the `weightVector` will really clears the weightVector of the `cf`. Are you sure this is what I wanted to write? (I had a hard time to debug this part and make the changes that I want into effect. )